### PR TITLE
Odsc 45131/ads interfaces

### PR DESCRIPTION
**Pull request by guptadivyank**
add dataset.get_features_df  and dataset.get_features interfaces
update documentation and minor bug fix

### CODE REVIEW

File: ads/feature_store/dataset.py [hunk 1]
```diff
@@ -733,7 +767,18 @@ def _update_from_oci_dataset_model(self, oci_dataset: OCIDataset) -> "Dataset":
 
         for infra_attr, dsc_attr in self.attribute_map.items():
             if infra_attr in dataset_details:
-                self.set_spec(infra_attr, dataset_details[infra_attr])
+                if infra_attr == self.CONST_OUTPUT_FEATURE_DETAILS:
+                    # May not need if we fix the backend and add feature_group_id to the output_feature
+                    features_list = []
+                    for output_feature in dataset_details[infra_attr]["items"]:
```
Comment (najiyacl) [inline ads/feature_store/dataset.py:773]: why is this needed?Does backend expect this?

File: ads/feature_store/dataset.py [hunk 2]
```diff
@@ -733,7 +767,18 @@ def _update_from_oci_dataset_model(self, oci_dataset: OCIDataset) -> "Dataset":
 
         for infra_attr, dsc_attr in self.attribute_map.items():
             if infra_attr in dataset_details:
-                self.set_spec(infra_attr, dataset_details[infra_attr])
+                if infra_attr == self.CONST_OUTPUT_FEATURE_DETAILS:
+                    # May not need if we fix the backend and add feature_group_id to the output_feature
```
Comment (najiyacl) [inline ads/feature_store/dataset.py:771]: nit:

File: ads/feature_store/dataset.py (hunk 1 above)
Comment (yogesh266) [reply ads/feature_store/dataset.py:773]: No actually for query interface we will need datasetId. as of now we are providing the query interface on dataset but in future we might have similar to featuregroup so i added it.
it will be only related to ads not backend.

File: ads/feature_store/mixin/oci_feature_store.py [hunk 3]
```diff
@@ -27,3 +28,44 @@ def init_client(
     @property
     def client(self) -> oci.feature_store.feature_store_client.FeatureStoreClient:
         return super().client
+
+    @class_or_instance_method
+    def list_resource(
```
Comment (KshitizLohia) [inline ads/feature_store/mixin/oci_feature_store.py:33]: can we add integration test for list_resource method

File: ads/feature_store/dataset.py [hunk 4]
```diff
@@ -690,6 +702,28 @@ def delete(self):
 
         dataset_execution_strategy.delete_dataset(self, dataset_job)
 
+    def get_features(self) -> List[DatasetFeature]:
```
Comment (KshitizLohia) [inline ads/feature_store/dataset.py:705]: do we have `def get_last_job(self)` for dataset?

File: ads/feature_store/feature_group.py [hunk 5]
```diff
@@ -921,10 +916,18 @@ def get_last_job(self) -> "FeatureGroupJob":
             )
 
         if not self.job_id:
-            raise ValueError(
```
[the remark this answers is not in the document]
Comment (najiyacl) [reply ads/feature_store/feature_group.py:924]: we can use get_last_job for stat and validation.

**Review by KshitizLohia — APPROVED** [review]
looks good to me. please attach the team city build before merge and backmerge with `feature/featurestore/1.0.3`

File: ads/feature_store/mixin/oci_feature_store.py (hunk 3 above)
Comment (guptadivyank) [reply ads/feature_store/mixin/oci_feature_store.py:33]: Done

**Comment by guptadivyank** [comment]
https://teamcity.oci.oraclecorp.com/buildConfiguration/Odsc_FeatureStore_FeatureStoreAdsIntegrationTests_FeatureStoreAdsIntegrationTestsPython38/41597143?buildTab=buildParameters&onlyModified=true Integration tests passed
